packages/components/src/components/data-list/data-list-row.tsx

```diff
@@ -48,11 +49,36 @@ const DataListRow = ({
     const [show_desc, setShowDesc] = React.useState(false);
     const isMounted = useIsMounted();
 
+    const debouncedHideDetails = React.useMemo(
+        () =>
+            debounce(() => {
+                setShowDesc(false);
+            }, 5000),
+        []
+    );
+
+    const toggleDetails = () => {
+        if (action_desc) {
+            setShowDesc(!show_desc);
+            debouncedHideDetails();
+        }
+    };
+
+    const toggleDetailsDecorator = (e?: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) => {
+        clickAndKeyEventHandler(toggleDetails, e);
```

Comment:
`clickAndKeyEventHandler` will check if event was from keyboard or mouse and modify behaviour accordingly. Because of that `toggleDetailsDecorator` can be used for both onClick and onKeyDown. 
If I don't do this I'll have a sonarCload issue, that non-interactive element (`<div>`) has only onClick.